crates/iota-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -1581,4 +1632,213 @@ mod object_cost_tests {
+
+        // Read/write to object 1 should go through even though the budget is exceeded
+        // even before the cost of this tx is considered.
+        for mutable in [true, false].iter() {
+            let tx = build_transaction(
+                &[(shared_obj_1, *mutable)],
+                tx_gas_budget,
+                TEST_ONLY_GAS_PRICE,
+            );
+            if let SequencingResult::Schedule(_) = shared_object_congestion_tracker.try_schedule(
+                &tx,
+                max_execution_duration_per_commit,
+                max_overshoot_per_commit,
+                &HashMap::new(),
+                0,
+            ) {
+                // pass
+            } else {
+                panic!("should schedule");
+            }
+        }
+
+        // Transactions touching both objects should be deferred, with object 0 as the
+        // congested object.
+        for mutable_0 in [true, false].iter() {
+            for mutable_1 in [true, false].iter() {
+                let tx = build_transaction(
+                    &[(shared_obj_0, *mutable_0), (shared_obj_1, *mutable_1)],
+                    tx_gas_budget,
+                    1,
+                );
+                if let SequencingResult::Defer(_, congested_objects) =
+                    shared_object_congestion_tracker.try_schedule(
+                        &tx,
+                        max_execution_duration_per_commit,
+                        max_overshoot_per_commit,
+                        &HashMap::new(),
+                        0,
+                    )
+                {
+                    if assign_min_free_execution_slot {
+                        assert_eq!(congested_objects.len(), 2);
+                    } else {
+                        assert_eq!(congested_objects.len(), 1);
+                        assert_eq!(congested_objects[0], shared_obj_0);
+                    }
+                } else {
+                    panic!("should defer");
+                }
+            }
+        }
+    }
+
+    #[rstest]
+    fn test_accumulated_debts(
+        #[values(
+            PerObjectCongestionControlMode::TotalGasBudget,
+            PerObjectCongestionControlMode::TotalTxCount
+        )]
+        mode: PerObjectCongestionControlMode,
+        #[values(true, false)] assign_min_free_execution_slot: bool,
+    ) {
+        // Creates two shared objects and three transactions that operate on these
+        // objects.
+        let shared_obj_0 = ObjectID::random();
+        let shared_obj_1 = ObjectID::random();
+
+        let tx_gas_budget = 100;
+
+        // Set max_accumulated_txn_cost_per_object_in_commit to only allow 1 transaction
+        // to go through before overage occurs.
+        let max_execution_duration_per_commit = match mode {
+            PerObjectCongestionControlMode::None => unreachable!(),
+            PerObjectCongestionControlMode::TotalGasBudget => 90,
+            PerObjectCongestionControlMode::TotalTxCount => 2,
```

Comment:
```suggestion
            PerObjectCongestionControlMode::TotalTxCount => 1,
```

Why is it 2 if the comment says "to allow 1 transaction to go through"?

Reply:
Because the initial debts are set to 2, so a single transaction will result in an overshoot.
I rephrased it to highlight the fact that the initial debts play a role as well and made the phrasing more clear.